src/main/java/de/tum/cit/aet/artemis/programming/service/localvc/LocalVCPrePushHook.java

```diff
@@ -73,10 +73,19 @@ public void onPreReceive(ReceivePack receivePack, Collection<ReceiveCommand> com
             return;
         }
 
-        // Reject pushes to anything other than the default branch.
-        if (!command.getRefName().equals("refs/heads/" + defaultBranchName)) {
-            command.setResult(ReceiveCommand.Result.REJECTED_OTHER_REASON, "You cannot push to a branch other than the default branch.");
-            return;
+        // If branching is allowed, reject pushes to anything other than branch names that match the branch regex
+        // If branching is not allowed, reject pushes to anything other than the default branch
+        String branchPath = command.getRefName();
+        String branchName = branchPath.substring("refs/heads/".length());
+        int branchingAllowed = localVCServletService.isBranchNameAllowedForRepository(repository, branchName);
```

Comment:
Using an `int` as return error code is a strong C-ism that is very unusal in Java.

Possible solution (first one that came to mind, not necessarily the best one): Could you define an inner enum class in the service with the three options and then switch-pattern-match over them here?